app/src/main/java/gateway/controller/CtrlShareFile.java

```diff
@@ -0,0 +1,78 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.services.ServiceAuth.ResUUID;
+import gateway.services.UtilValidator;
+import gateway.soap.request.ReqShareFile;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONObject;
+
+public class CtrlShareFile
+{
+	public static ResStatus share_file (ReqShareFile args)
+	{
+		ResStatus statusRes = new ResStatus ();
+
+		// validations of all fields
+		ResStatus resValidate = UtilValidator.validate (args);
+		if (resValidate.error) {
+			return ResStatus.downCast (ResStatus.class, resValidate);
+		}
+
+		// validation of auth
+		ResStatus resAuth = ServiceAuth.authenticate (args.token);
+		if (resAuth.error) {
+			return ResStatus.downCast (ResStatus.class, resAuth);
+		}
+
+		// obtain uuid from user and otheruser
+		UUID userUUID = UUID.fromString (ServiceAuth.tokenGetClaim (args.token, "uuid"));
+		ResUUID otherUserUUID = ServiceAuth.getUserUUID (args.token, args.otherUsername);
```

Comment:
ResUUID inherits from ResStatus, so it's not guaranteed that the uuid field will be filled. 

Add a check to ensure ResUUID didn't complete with errors. (otherUserUUID.error)